test/common_nn.py

```diff
@@ -217,6 +217,35 @@
 ]
 
 
+functional_criterion_tests = [
```

Comment:
I think these functions can go in `test_nn.py`, because I believe `common_nn` is supposed to contain functions that should be run both in `nn` and `legacy_nn`.